runtime/server/generate_metrics_view.go

```diff
@@ -94,12 +95,32 @@ func (s *Server) GenerateMetricsViewFile(ctx context.Context, req *runtimev1.Gen
 	var data string
 	var aiSucceeded bool
 	if req.UseAi {
-		data, err = s.generateMetricsViewYAMLWithAI(ctx, req.InstanceId, olap.Dialect().String(), req.Connector, tbl.Name, isDefaultConnector, model != nil, tbl.Schema)
+		// Generate
+		start := time.Now()
+		res, err := s.generateMetricsViewYAMLWithAI(ctx, req.InstanceId, olap.Dialect().String(), req.Connector, tbl.Name, isDefaultConnector, model != nil, tbl.Schema)
 		if err != nil {
 			s.logger.Warn("failed to generate metrics view YAML using AI", zap.Error(err))
 		} else {
+			data = res.data
 			aiSucceeded = true
 		}
+
+		// Emit event
+		attrs := []attribute.KeyValue{attribute.Int("table_column_count", len(tbl.Schema.Fields))}
+		attrs = append(attrs,
+			attribute.Bool("succeeded", aiSucceeded),
+			attribute.Int64("elapsed_ms", time.Since(start).Milliseconds()),
+		)
+		if res != nil {
+			attrs = append(attrs,
+				attribute.Int("valid_measures_count", res.validMeasures),
+				attribute.Int("invalid_measures_count", res.invalidMeasures),
+			)
+		}
+		if err != nil {
+			attrs = append(attrs, attribute.String("error", err.Error()))
+		}
+		s.activity.Record(ctx, activity.EventTypeLog, "ai_generated_metrics_view_yaml", attrs...)
```

Comment:
This might be recorded as a metric and allowed in a `filterSink` as
```
		sink := activity.NewFilterSink(intakeSink, func(e activity.Event) bool {
			return e.EventType != activity.EventTypeMetric || name == "ai_generated_metrics_view_yaml_elapsed_ms"
		})
```
not very elegant as it requires to maitain a list of allowed metric names. but ideally, we anyway need to create a list of constants that represent all registered application metric names so that it is possible to quickly check if there are any duplicate names
so it is up to you, i'm fine with the existing solution